src/pkg/repository/repository.go

```diff
@@ -633,16 +633,26 @@ func getBackupErrors(
 // DeleteBackups removes the backups from both the model store and the backup
 // storage.
 //
+// If failOnMissing is true then returns an error if a backup model can't be
+// found. Otherwise ignores missing backup models.
+//
+// Missing models or snapshots during the actual deletion do not cause errors.
+//
 // All backups are delete as an atomic unit so any failures will result in no
 // deletions.
-func (r repository) DeleteBackups(ctx context.Context, ids ...string) error {
-	return deleteBackups(ctx, store.NewWrapper(r.modelStore), ids...)
+func (r repository) DeleteBackups(
+	ctx context.Context,
+	failOnMissing bool,
+	ids ...string,
+) error {
+	return deleteBackups(ctx, store.NewWrapper(r.modelStore), failOnMissing, ids...)
 }
 
-// deleteBackup handles the processing for Backup.
+// deleteBackup handles the processing for backup deletion.
 func deleteBackups(
 	ctx context.Context,
 	sw store.BackupGetterModelDeleter,
+	failOnMissing bool,
```

Comment:
While I'm fine with this implementation, it does make me wonder if it's the most idiomatic.  Another way of handling batch processes like this that I've liked is to return an `[]error` of `len=len(ids)`.  Every entry in the returned slice relates to the same-index of the id that it processed, holding nil if no error occurred.  So for 3 ids, the response may be `[]error{nil, ErrNotFound, nil}`.  This way the caller can: 1/ easily associate errors with IDs. 2/ respond to each error holistically (what if we want to continue on some other error case as well?).

Just spitballing here.  Changes aren't necessary for this PR unless you want the refactor.